absl/base/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+#
+# Copyright 2017 The Abseil Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+file(GLOB BASE_PUBLIC_HEADERS "*.h")
+file(GLOB BASE_INTERNAL_HEADERS "internal/*.h")
+
+# base main library
+list(APPEND BASE_SRC 
+    "internal/cycleclock.cc"
+    "internal/raw_logging.cc"
+    "internal/spinlock.cc"
+    "internal/sysinfo.cc"
+    "internal/thread_identity.cc"
+    "internal/unscaledcycleclock.cc"
+    "internal/spinlock_wait.cc"
+    "internal/low_level_alloc.cc"
+    "internal/malloc_hook.cc"
+    "dynamic_annotations.cc"
+    ${BASE_PUBLIC_HEADERS}
+    ${BASE_INTERNAL_HEADERS}
+)
+
+abseil_library(base)
+
+# malloc extension library
+set(MALLOC_EXTENSION_SRC "internal/malloc_extension.cc")
+set(MALLOC_EXTENSION_PUBLIC_LIBRARIES base)
+
+abseil_library(malloc_extension)
+
+# throw delegate library
+set(THROW_DELEGATE_SRC "internal/throw_delegate.cc")
+
+abseil_library(throw_delegate)
+
+
+install(FILES ${BASE_PUBLIC_HEADERS}
```

Comment:
Couldn't these INSTALL commands be handled by `abseil_library` and by passing the headers in as an argument?

Reply:
Some abseil components are header-only and cmake 2.8 does not support yet target-interface with header only content. I find more consistent to use install() for every headers and to avoid to mix both. If we were using cmake 3.0, things would be different